pkgs/applications/networking/irc/weechat/wrapper.nix

```diff
@@ -0,0 +1,29 @@
+let
+  weechatDefault = "$HOME/.weechat";
+in
+  { weechat, plugins ? {}, buildFHSUserEnv
+  , homeDir ? weechatDefault, term ? "xterm" }:
+
+  let
+    weechat' =
+      if homeDir == weechatDefault
+      then weechat
+      else weechat.override { inherit homeDir; };
+
+    termScript = ''
+      export TERM=${term}
+    '';
+
+    pluginScripts = map (a: ''
+      mkdir -p ${homeDir}/$(dirname ${a})
+      cp -f ${plugins."${a}"}/share/$(basename ${a}) ${homeDir}/${a}
+    '') (builtins.attrNames plugins);
+  in
+    buildFHSUserEnv {
```

Comment:
I have not yet understood, how `buildFHSUserEnv` is useful here.

Reply:
The idea was inspired by the following article: http://sandervanderburg.blogspot.de/2013/09/composing-fhs-compatible-chroot.html

WeeChat modifies its HOME directory during runtime, so the HOME directory must live in an imperative/writable environment.

In the previous PR of mine I packaged a script for WeeChat and @fpletz asked for an approach to load this into WeeChat. Now I'm trying to build an environment here which can be used by Nix to load such packaged scripts into it, but is writable as well due to the runtime behavior of WeeChat.

If you know a better approach, I'm happy to discuss about it :-)

Reply:
I'm not entirely sure about this, but would this pull request offer a better(?) solution to the problem that this FSH approach is solving: https://github.com/NixOS/nixpkgs/pull/25274 Or are they solving different problems?

Reply:
no, this was just an experiment to get Weechat running with nix-based plugins and to get some feedback (and maybe integrate this into nixpkgs).
I didn't see the efforts in #25274, however this seems to be quite interesting and I'd close this as soon as this is merged.